src/main/java/Task.java

```diff
@@ -0,0 +1,33 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "[X]" : "[ ]"); //mark done task with x
+    }
+
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    public void markAsUndone() {
+        this.isDone = false;
+    }
+
+    public void fullDesc() {
+        System.out.println("      " + this.getStatusIcon() + " " + this.description);
+    }
+
+    public String stringDesc() {
+        return this.getStatusIcon() + " " + this.description;
+    }
+
+    public String textDesc() {
+        return this.description;
+    }
```

Comment:
We could try to override the `toString` method of the class to retrieve the string representation of the task.